lib/src/hooks/useFunctionRef.ts

```diff
@@ -0,0 +1,11 @@
+import * as React from "react";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const useFunctionRef = <Fn extends (...args: any[]) => any>(fn?: Fn) => {
```

Comment:
Could we change this to:
`Fn extends (...args: Parameters<Fn>) => ReturnType<Fn>`?

Reply:
That looks like a recursive structure to me? This is how I typically do generic functions, and see them in the wild – ESLint is just really aggressive about these `any`s.

Reply:
Yeah, thinking of how we can do this without any. But maybe that's the play here.